File: ios/boxes/CacheImage/RNTCacheImageManager.m
//
//  RNTCacheImageManager.m
//  boxes
//
//  Created by Pavlo Huk on 30.01.2022.
//

#import <React/RCTViewManager.h>
 
@interface RCT_EXTERN_MODULE(RNTCacheImageManager, RCTViewManager)
 
RCT_EXPORT_VIEW_PROPERTY(uri, NSString)
RCT_EXPORT_VIEW_PROPERTY(headers, NSDictionary)

@end

File: ios/boxes/CacheImage/RNTCacheImageManager.swift
//
//  RNTCacheImageManager.swift
//  boxes
//
//  Created by Pavlo Huk on 30.01.2022.
//

import UIKit
import SDWebImage

@objc (RNTCacheImageManager)
class RNTCacheImageManager: RCTViewManager {
  
  override init() {
    super.init()

    SDWebImageManager.shared.cacheKeyFilter = SDWebImageCacheKeyFilter { uri in
      let url = uri.path.components(separatedBy: "?").first?.description
      let key = url?.components(separatedBy: "/").last?.description
      
      return key ?? uri.absoluteString
    }
  }
 
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
 
  override func view() -> UIView! {
    return CacheImageView()
  }
}
